ale_linters/mercury/mmc.vim

```diff
@@ -0,0 +1,44 @@
+" Author: stewy33 <slocumstewy@gmail.com>
+" Description: Lints mercury files using mmc
+
+if !exists('g:ale_mercury_mmc_executable')
+    let g:ale_mercury_mmc_executable = 'mmc'
+endif
+
+if !exists('g:ale_mercury_mmc_options')
+    let g:ale_mercury_mmc_options = ''
+endif
+
+function! ale_linters#mercury#mmc#GetCommand(buffer) abort
+    return ale#Var(a:buffer, 'mercury_mmc_executable')
+    \ . ' --errorcheck-only '
+    \ . ale#Var(a:buffer, 'mercury_mmc_options')
+    \ . ' ' . expand('#' . a:buffer . '...')
+endfunction
+
+function! ale_linters#mercury#mmc#Handle(buffer, lines) abort
+    " output format
+    " <filename>:<line>:   <issue type>: <message>
+    let l:pattern = '\v^\w+\.m:(\d+):\s+((\warning|\wrror)): (.*)'
+    let l:output = []
+
+    for l:match in ale#util#GetMatches(a:lines, l:pattern)
+        let l:type = tolower(l:match[2])
+
+        call add(l:output, {
+        \   'lnum': l:match[1],
+        \   'type': l:type is? 'warning' ? 'W' : 'E',
+        \   'text': l:type . ': ' . l:match[4],
+        \})
+    endfor
+
+    return l:output
+endfunction
```

Comment:
Add a test for this function in the `test/handler` directory, which contains a few examples.